### ISSUE TITLE
Code Cleanup - termDispatch.hpp & adaptDispatch.hpp overrides 

### ISSUE THREAD

**Issue by dlong11** [microsoft/terminal#896]
Minor code cleanup

The ITermDispatch overridden methods in terminal/src/terminal/adapter/termDispatch.hpp & terminal/src/terminal/adapter/adaptDispatch.hpp do not follow 
[C.128: Virtual functions should specify exactly one of virtual, override, or final guideline. ](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final)

**Comment by dlong11** [comment]
I just noticed that InteractDispatch.hpp has the same issue. (in this case, it is the IInterractDispatch overrides).   I can modify this issue or create a new issue.  Let me know what you think is best.   

Side note - I noticed these when doing a code review.   I am assuming the Core guideline analyzer isn't being run as part of the CI process.   Is the plan to do this?

**Comment by zadjii-msft** [comment]
You're right, they probably shouldn't be `virtual`.

I believe we never turned the analyzer on because so much of the code was written before we started conforming to that standard, that we'd have spent entire releases just cleaning it up. @adiviness can comment more.

**Comment by adiviness** [comment]
Currently we don't have a static analyzer looking at the cpp core check stuff. In `/src` there is a ruleset file that I think is empty and it was only set up on the `/buffer/out` directory when the `AuditCode` configuration was specified.

**Comment by DHowett-MSFT** [comment]
This was fixed in #1004.
